src/lex/cpp.rs

```diff
@@ -590,6 +594,7 @@ impl<'a> PreProcessor<'a> {
         let location = self.span(start);
 
         // TODO: is this unwrap safe? there should only be scalar types in a cpp directive...
+        self.update_builtin_definitions();
```

Comment:
Same comment, I think this only needs to be done just before `replace()`.

Reply:
I put it here because `cpp_expr` is a free function so I can't put it in front of the internal `replace`. However, `cpp_expr` only looks at one line so nothing should change. I should put it before the TODO comment though.